### PR TITLE
.gitignore improvements for Windows/macOS

### PR DESCRIPTION
**Pull request by mhilbrunner**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,6 @@ platform/android/java/local.properties
 platform/android/java/project.properties
 platform/android/java/libs/*
 platform/android/java/assets
-platform/android/java/.idea/*
```
Comment (mhilbrunner) [inline .gitignore]: Moved to top level, similar to other IDEs, as Jetbrains has products focused on C/C++, Python, and C#

**Review by akien-mga — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -245,8 +251,11 @@ __pycache__/
 #Kdevelop project files
 *.kdev4
 
-# xCode
-xcuserdata
+# Xcode
+xcuserdata/
+*.xcscmblueprint
+*.xccheckout
+*.xcodeproj/*
```
Comment (Calinou) [inline .gitignore]: Wouldn't `*.xcodeproj/` be enough?

Comment (mhilbrunner) [reply .gitignore]: In the Xcode projects I have around it seems that way, although I'm on the most recent version which seems to behave a little better and I don't use Xcode VCS/checkout or blueprint features enough to be certain.

These were adapted from https://github.com/github/gitignore/blob/master/Global/Xcode.gitignore while omitting the pre-Xcode-8 compat entries and the last section, which IMO is mostly needed in *actual* Xcode iOS projects. (I.e. where you want to share certain IDE stuff in your team, which shouldn't matter for us.)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**